network/ieee802_11_prism.ksy

```diff
@@ -0,0 +1,63 @@
+meta:
+  id: ieee802_11_prism
+  title: prism II headers for 802.11 capture
+  license: Unlicense
+  endian: le
+  encoding: ascii
+
+doc-ref:
+  - https://www.tcpdump.org/linktypes/LINKTYPE_IEEE802_11_PRISM.html
+
+seq:
+  - id: message_code
+    type: u4
+  - id: size
+    type: u4
+  - id: device_name
+    type: strz
+    size: 16
+  - id: fields
+    type: fields
+    size: fields_size
+  - id: body
+    size-eos: true
+instances:
+  header_size:
+    value: 4 + 4 + 16
+  fields_size:
+    value: size - header_size
+types:
+  fields:
+    seq:
+      - id: fields
+        type: field
+        repeat: eos
+    types:
+      field:
+        seq:
+          - id: flags
+            type: u1
+          - id: type
+            type: u1
+            enum: type
+          - id: unknown
+            type: u2
+            doc: usually zero
+          - id: is_supplied
+            type: u2
+          - id: size
+            type: u2
+          - id: value
+            type: u4
```

Comment:
This violates the spec as well. It says

> The Length field **indicates the length** of the following **Data** field; in theory, **it could have any value**, but it is almost always, if not always, 4. The Data field contains the value of the parameter.

Despite that it is _almost always 4_, it's better to not rely on it, and behave according to what the spec says with certainty: indicates the length of the following Data field.

If the `field` type was enclosed in a substream with a fixed size, I would still say "well, I can't read the full data of this single `field`, because it's more than 4, but at least I can access the following `field`s", but this `field` type uses the parent substream, so ignoring the `size` can quickly lead to malformed data and almost certainly with an EOF exception at the end of the file.

Reply:
I know very well that you don't like this kind of stuff, but still please consider renaming `size` to `len_value` (as per the style guide).

Reply:
Usage of length is fixed.
Change of the name isn't.

>I know very well that you don't like this kind of stuff, but still please consider renaming size to len_value (as per the style guide).

I'm sorry, but IMHO it's the style guide should be fixed instead of just following the badly-designed style guide.

Reply:
@KOLANICH:
> I'm sorry, but IMHO it's the style guide that should be fixed instead of just following the badly-designed style guide.

Why exactly do you think it is badly designed? IMHO introducing a naming convention like this brings consistency and order and improves orientation in the spec. When you see an attribute starting with `len_`, you know immediately that it's a size in bytes of some other attribute, and that attribute is named exactly as what follows after the `len_` prefix.

If you have a problem with the particular `len_` prefix and you'd rather like `size_` for example, we can talk about such change, but I don't think it matters. It's just nice to have some established convention, rather than for everyone to name it according to the current mood.

Reply:
Starting an attr with `len_` or any other generic prefix means I cannot meaningfully use editor autocompletion.